workflows/prognostic_c48_run/runtime/steppers/radiation.py

```diff
@@ -0,0 +1,221 @@
+            sfc_filename,
+            sfc_data,
+        )
+
+    def __call__(
+        self, time: cftime.DatetimeJulian, state: State,
+    ):
+        self._rad_update(time, self._timestep)
+        if self._input_generator is not None:
+            state = self._generate_inputs(state, time)
+        diagnostics = self._rad_compute(state, time)
+        return {}, diagnostics, {}
+
+    def _rad_update(self, time: cftime.DatetimeJulian, dt_atmos: float) -> None:
+        """Update the radiation driver's time-varying parameters"""
+        # idat is supposed to be model initalization time but is unused w/ current flags
+        idat = np.array(
+            [time.year, time.month, time.day, 0, time.hour, time.minute, time.second, 0]
+        )
+        jdat = np.array(
+            [time.year, time.month, time.day, 0, time.hour, time.minute, time.second, 0]
+        )
+        fhswr = np.array(float(self._rad_config["fhswr"]))
+        dt_atmos = np.array(float(dt_atmos))
+        aerosol_data = io.load_aerosol(self._forcing_local_dir)
+        _, solar_data = io.load_astronomy(
+            self._forcing_local_dir, self._rad_config["isolar"]
+        )
+        gas_data = io.load_gases(self._forcing_local_dir, self._rad_config["ictmflg"])
+        self._driver.radupdate(
+            idat,
+            jdat,
+            fhswr,
+            dt_atmos,
+            self._rad_config["lsswr"],
+            aerosol_data["kprfg"],
+            aerosol_data["idxcg"],
+            aerosol_data["cmixg"],
+            aerosol_data["denng"],
+            aerosol_data["cline"],
+            solar_data,
+            gas_data,
+        )
+
+    def _rad_compute(self, state: State, time: cftime.DatetimeJulian,) -> Diagnostics:
+        """Compute the radiative fluxes"""
+        statein = preprocessing.statein(
+            state, self._tracer_inds, self._rad_config["ivflip"]
+        )
+        grid, coords = preprocessing.grid(state)
+        sfcprop = preprocessing.sfcprop(state)
+        ncolumns, nz = statein["tgrs"].shape[0], statein["tgrs"].shape[1]
+        model = preprocessing.model(
+            self._rad_config,
+            self._tracer_inds,
+            time,
+            self._timestep,
+            nz,
+            self._comm.rank,
+        )
+        random_numbers = io.generate_random_numbers(
+            ncolumns, nz, radiation.NGPTSW, radiation.NGPTLW
+        )
+        lw_lookup = io.load_lw(self._lookup_local_dir)
+        sw_lookup = io.load_sw(self._lookup_local_dir)
+        out = self._driver._GFS_radiation_driver(
+            model, statein, sfcprop, grid, random_numbers, lw_lookup, sw_lookup
+        )
+        out = preprocessing.postprocess_out(out)
+        return preprocessing.unstack(out, coords)
+
+    def _generate_inputs(self, state: State, time: cftime.DatetimeJulian) -> State:
+        if self._input_generator is not None:
+            _, _, state_updates = self._input_generator(time, state)
```

Comment:
I'm finding it hard to follow the indirection here. First, I'd recommend inlining `_generate_inputs` where it is called and deleting this function. Also, what is the _input_generator and is it really necessary to have the  complex/untested`MergedState` class? 

The pattern seems roughly similar to this https://github.com/ai2cm/fv3net/blob/c4e7f17d25bd97c83ba291edd8a6ef7afc451510/workflows/prognostic_c48_run/runtime/transformers/core.py#L35 and perhaps more elegantly handled as function decorator.